### ISSUE TITLE
Gradle no method 'register()' for printClasspath

### ISSUE THREAD

**Issue by lukaswoellhaf** [CodeIntelligenceTesting/cifuzz#550]
I'm using cifuzz inside a gradle project. When trying to run the fuzzing tests I'm getting following error:

```
FAILURE: Build failed with an exception.

* Where:
Initialization script '/Users/<username>/.local/share/cifuzz/share/cifuzz/share/init.gradle' line: 9

* What went wrong:
Could not find method register() for arguments [printClasspath, init_3nwb7i25vko1ydtq2x5p6v36f$_run_closure1$_closure3@34739756] on task set of type org.gradle.api.internal.tasks.DefaultTaskContainer.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
```
gradlew: exit status 1

Which leads to following:

``` Gradle
tasks.register("printClasspath") {
    println "cifuzz.test.classpath=${sourceSets.test.runtimeClasspath.asPath}";
}

tasks.register("printBuildDir") {
    println "cifuzz.test.buildDir=${layout.buildDirectory.get()}";
}
```

Is there a possibility that my build.gradle file interferes the init.gradle file for cifuzz, and how could I solve this?

**Comment by jochil** [comment]
Thanks for the issue

> Is there a possibility that my build.gradle file interferes the init.gradle file for cifuzz, and how could I solve this?

I think this can be, but I also need to dive a little bit deeper. Can you provide some insight in your `build.gradle`? Maybe a stripped down version or some parts of it?
And which gradle version are you using?

**Comment by lukaswoellhaf** [comment]
I used Gradle version `4.8.1` which is as I notice now not compatible wit cifuzz. I upgraded to Gradle version `4.9` and the issue mentioned above resolved.
But unfortunately a new issue with the `init.gradle` file  occurs:

``` bash
❌ 
FAILURE: Build failed with an exception.

* Where:
Initialization script '/Users/lwoellhaf/.local/share/cifuzz/share/cifuzz/share/init.gradle' line: 35

* What went wrong:
Could not create task 'cifuzzTest' (Test)
> Selected test name cannot be null or empty.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 0s
gradlew: exit status 1
```

Which leads to:
``` gradle
tasks.register("cifuzzTest", Test) {
        useJUnitPlatform()
        ignoreFailures = true
        // disable jazzer hooks because they are not needed for coverage runs
        jvmArgs '-Djazzer.hooks=false'

        jacoco {
            destinationFile = execFile
        }
        filter {
            includeTestsMatching fuzzTest
        }
    }
```
I'm guessing this also could resolve with a more recent gradle version? I will try it out.

**Comment by jochil** [comment]
Do you have already a JUnit setup in your `build.gradle`? What we are doing here is extending the existing `Test` task from JUnit, IIRC

Usually this should be done by adding "JUnit" to the `build.gradle... you can see it in our gradle example: https://github.com/CodeIntelligenceTesting/cifuzz/blob/main/examples/gradle/build.gradle

**Comment by lukaswoellhaf** [comment]
Here is part of my `build.gradle` file. Yes I already had the JUnit set up before

``` gradle
//Some Code
dependencies {
//Some Code

	// This dependency provides the mocking framework Mockito.
	testCompile 'org.mockito:mockito-inline:4.6.1'
	testImplementation 'org.mockito:mockito-junit-jupiter:4.6.1'

	// This dependencies provide the JUnit5 testing framework.
	testImplementation 'org.junit.jupiter:junit-jupiter:5.9.0'
	testImplementation('org.junit.jupiter:junit-jupiter-api:5.9.0')
	testRuntimeOnly('org.junit.platform:junit-platform-launcher:1.9.0')
	testImplementation('org.junit.jupiter:junit-jupiter-engine:5.9.0')

	// This dependency provides backwards compatibility with JUnit4 and running Groovy-Tests.
	testRuntimeOnly('org.junit.vintage:junit-vintage-engine:5.9.0')

	// This dependency provides the CLI fuzz tool cifuzz
	testImplementation 'com.code-intelligence:jazzer-junit:0.13.2'
}

        // This section is needed to run JUnit5 Tests
        test {
	        useJUnitPlatform()
        }

//Some Code
```

**Comment by jochil** [comment]
I was able to reproduce the error but unfortunately gradle 7.5 is the first version that runs without errors. 
But I gained some more insights and think it is possible to get it working with older gradle versions, but this will require some work and a new release... so it might take a while. I will keep this issue updated

**Comment by jochil** [comment]
I created a (draft) pr that should get it working... at least with gradle 5.0

@lukaswoellhaf it would be great if you can verify that it solved your issue by using a pre-release installer from here: https://github.com/CodeIntelligenceTesting/cifuzz/actions/runs/3830002556
just unpack the archive and execute the binary for you OS

**Comment by lukaswoellhaf** [comment]
@jochil Can confirm. Cifuzz is now compatible with `Gradle 5.0`. Everything below is still incompatible (for example `Gradle 4.10`), but this is expected and since the upgrade not an issue for me anymore. Thank you very much you made my day 👑

**Comment by jochil** [comment]
great... happy to hear that :) It will go into the next release

**Comment by jochil** [comment]
The fix is included in the current release 0.14.0: https://github.com/CodeIntelligenceTesting/cifuzz/releases/tag/v0.14.0
